experiments/axon_geometry/notebooks/histograms_from_root.ipynb

```diff
@@ -0,0 +1,315 @@
+    "\n",
+    "log_d_from_root = np.log10(d_from_root)\n",
+    "L = len(log_d_from_root)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compute histogram of the curvatures (uses GPU)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "zero_curvatures_log_d_from_root = log_d_from_root[np.where(curvatures < 1e-16)[0]]\n",
+    "nonzero_curvatures_log_d_from_root = log_d_from_root[np.where(curvatures > 1e-16)[0]]\n",
+    "# compute histogram of zero-curvatures\n",
+    "torch_zero_curvatures_log_d_from_root = torch.from_numpy(zero_curvatures_log_d_from_root)\n",
+    "zero_curvatures_hist_min = torch.min(torch_zero_curvatures_log_d_from_root)\n",
+    "zero_curvatures_hist_max = torch.max(torch_zero_curvatures_log_d_from_root)\n",
+    "zero_curvatures_hist_bin_edges = torch.arange(\n",
+    "    zero_curvatures_hist_min,\n",
+    "    zero_curvatures_hist_max,\n",
+    "    (zero_curvatures_hist_max - zero_curvatures_hist_min)/bins)\n",
+    "\n",
+    "zero_curvatures_hist = torch.histc(\n",
+    "    torch_zero_curvatures_log_d_from_root,\n",
+    "    bins=bins,\n",
+    "    min=zero_curvatures_hist_min,\n",
+    "    max=zero_curvatures_hist_max,\n",
+    ")\n",
+    "# compute histogram of non-zero-curvatures (uses GPU)\n",
+    "torch_nonzero_curvatures_log_d_from_root = torch.from_numpy(nonzero_curvatures_log_d_from_root)\n",
+    "nonzero_curvatures_hist_min = torch.min(torch_nonzero_curvatures_log_d_from_root)\n",
+    "nonzero_curvatures_hist_max = torch.max(torch_nonzero_curvatures_log_d_from_root)\n",
+    "nonzero_curvatures_hist_bin_edges = torch.arange(\n",
+    "    nonzero_curvatures_hist_min,\n",
+    "    nonzero_curvatures_hist_max,\n",
+    "    (nonzero_curvatures_hist_max - nonzero_curvatures_hist_min) / bins,\n",
+    ")\n",
+    "nonzero_curvatures_hist = torch.histc(\n",
+    "    torch_nonzero_curvatures_log_d_from_root,\n",
+    "    bins=bins,\n",
+    "    min=nonzero_curvatures_hist_min,\n",
+    "    max=nonzero_curvatures_hist_max\n",
+    ")\n",
+    "\n",
+    "alpha_zero_curvatures = len(zero_curvatures_log_d_from_root) / L\n",
+    "alpha_nonzero_curvatures = len(nonzero_curvatures_log_d_from_root) / L\n",
+    "print(alpha_zero_curvatures, alpha_nonzero_curvatures)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compute histogram of the torsions (uses GPU)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "zero_torsions_log_d_from_root = log_d_from_root[np.where(torsions < 1e-16)[0]]\n",
+    "nonzero_torsions_log_d_from_root = log_d_from_root[np.where(torsions > 1e-16)[0]]\n",
+    "# compute histogram of zero-torsions\n",
+    "torch_zero_torsions_log_d_from_root = torch.from_numpy(zero_torsions_log_d_from_root).to(device)\n",
```

Comment:
i get an error:
NameError: name 'device' is not defined

Could you double check that it works for you? Do you have any ideas of what this may be?

Reply:
Resolved, it was a leftover from when I used GPU support for the histograms, now removed because it can be handled by a CPU just fine.